packages/flutter/lib/src/material/search.dart

```diff
@@ -0,0 +1,416 @@
+  ///
+  /// See also:
+  ///
+  ///  * [AppBar.actions], the intended use for the return value of this method.
+  List<Widget> buildActions(BuildContext context);
+
+  /// The theme used to style the [AppBar].
+  ///
+  /// By default, a white theme is used.
+  ///
+  /// See also:
+  ///
+  ///  * [AppBar.backgroundColor], which is set to [ThemeData.primaryColor].
+  ///  * [Appbar.iconTheme], which is set to [ThemeData.primaryIconTheme].
+  ///  * [AppBar.textTheme], which is set to [ThemeData.primaryTextTheme].
+  ///  * [AppBar.brightness], which is set to [ThemeData.primaryColorBrightness].
+  ThemeData appBarTheme(BuildContext context) {
+    assert(context != null);
+    final ThemeData theme = Theme.of(context);
+    assert(theme != null);
+    return theme.copyWith(
+      primaryColor: Colors.white,
+      primaryIconTheme: theme.primaryIconTheme.copyWith(color: Colors.grey),
+      primaryColorBrightness: Brightness.light,
+      primaryTextTheme: theme.textTheme,
+    );
+  }
+
+  /// The current query string shown in the [Appbar].
+  ///
+  /// The user manipulates this string via the keyboard.
+  ///
+  /// If the user taps on a suggestion provided by [buildSuggestions] this
+  /// string should be updated to that suggestion via the setter.
+  String get query => _textEditingController.text;
+  set query(String value) {
+    assert(query != null);
+    _textEditingController.text = value;
+  }
+
+  /// Transition to show search results.
+  ///
+  /// If the user taps on a suggestion provided by [buildSuggestions] the
+  /// screen should typically transition to the page showing the search
+  /// results for the suggested query. This transition can be triggered
+  /// by calling this method.
+  ///
+  /// See also:
+  ///
+  ///  * [showSuggestions] to show the search suggestions again.
+  ///  * [currentBody] to learn whether the results or the suggestions are
+  ///    currently shown.
+  void showResults(BuildContext context) {
+    focusNode.unfocus();
+    _currentBody.value = SearchBody.results;
+  }
+
+  /// Transition to show the search suggestions and places the input focus
+  /// back into the search text field.
+  ///
+  /// If the results are currently shown this method can be used to go back
+  /// to showing the search suggestions.
+  ///
+  /// See also:
+  ///
+  ///  * [showResults] to show the search results.
+  ///  * [currentBody] to learn whether the results or the suggestions are
+  ///    currently shown.
+  void showSuggestions(BuildContext context) {
+    FocusScope.of(context).requestFocus(focusNode);
+    _currentBody.value = SearchBody.suggestions;
+  }
+
+  /// The body that is currently shown under the [AppBar] in the search page.
+  ///
+  /// Will be null if the search page is currently not shown.
+  ///
+  /// See also:
+  ///
+  ///  * [showResults] to show the search results.
+  ///  * [showSuggestions] to show the search suggestions.
+  SearchBody get currentBody => _currentBody.value;
```

Comment:
Do SearchDelegate subclasses need to see this? If they really wanted to, they could remember if they'd most recently called showSuggestions or showResults

Reply:
I'll make it private as well.